### PR TITLE
DOCSP-877 & DOCSP-879 - Parameterize filter and update saved pref.

### PR DESCRIPTION
**Pull request by jdestefano-mongo**
MUST BE MERGED WITH:
https://github.com/mongodb/docs/pull/3074

### CODE REVIEW

File: sphinxext/tabs.py [hunk 1]
```diff
@@ -24,18 +24,18 @@
 .. raw:: html
 
    <div class="tabs">
-     <ul class="tab-strip tab-strip--singleton" role="tablist">
-       {{ for tab in tabs sortLanguages }}
+     <ul class="tab-strip tab-strip--singleton" role="tablist" PREFERENCE>
+       {{ for tab in tabs FILTER }}
```
Comment (jdestefano-mongo) [inline sphinxext/tabs.py]: Should we use some kind of notation with special characters or something to make it less like plain english? For example, $preference

Comment (i80and) [reply sphinxext/tabs.py]: I would go with e.g. `%PREFERENCE%`, yeah

File: themes/mongodb/src/js/componentTabs.js [hunk 2]
```diff
@@ -13,21 +13,60 @@ class TabsSingleton {
...
 
-    set languagePref(value) {
-        window.localStorage.setItem(this.key, value);
+    /**
+     * Sets the tabPref object depending on whether the tab belongs
+     * to set (e.g., "drivers") or if it's a one-off page.
+     * @param {object} value The "tabId" and optional "type" (tab set)
+     */
+    set tabPref(value) {
+        const tabPref = this.tabPref;
```
Comment (i80and) [inline themes/mongodb/src/js/componentTabs.js]: I'd not make this local variable

File: themes/mongodb/src/js/componentTabs.js [hunk 3]
```diff
@@ -13,21 +13,60 @@ class TabsSingleton {
...
-    get languagePref() {
-        return window.localStorage.getItem(this.key);
+    /**
+     * Return the tabPref object containing preferences for tab sets
+     * and page specific prefs. Returns an empty object if it doesn't
+     * exist.
+     * @returns {object} Tab preference object.
+     */
+    get tabPref() {
+        let pref = JSON.parse(window.localStorage.getItem(this.key));
```
Comment (i80and) [inline themes/mongodb/src/js/componentTabs.js]: Consider making this just
```js
return JSON.parse(window.localStorage.getItem(this.key)) || {};
```

Comment (jdestefano-mongo) [reply themes/mongodb/src/js/componentTabs.js]: Done

File: sphinxext/tabs.py (hunk 1 above)
Comment (jdestefano-mongo) [reply sphinxext/tabs.py]: Done

File: themes/mongodb/src/js/componentTabs.js (hunk 2 above)
Comment (jdestefano-mongo) [reply themes/mongodb/src/js/componentTabs.js]: Discussed offline, calling the getter method above.

**Pull request merged**